File: UI/src/app/models/question.ts
export interface Question {
    id: string;
    ask: string;
    answer: string;
    alternatives?: Array<string>;
    note?: string;
}
export interface QuestionData extends Question {
    value: number;
}

export interface QuestionPointer {
    id: string;
    value: number;
}

// TODO:  Depricate
export interface GameQuestion {
    questionId: string;
    value: number;
}

File: Api/TriviaGame.Api/TriviaGame.Api/Models/Question.cs
using System.ComponentModel.DataAnnotations;

namespace TriviaGame.Api.Models;

public class Question
{
    [Key]
    public string Id { get; set; }

    public string CategoryId { get; set; }
    public int Value { get; set; }
    public string Ask { get; set; }
    public string Answer { get; set; }
    public string? Note { get; set; }
    public string? Alternatives { get; set; }
}
